test/billiard_shake_and_bake_test.cpp

```diff
@@ -0,0 +1,91 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2025 Vissarion Fisikopoulos
+// Copyright (c) 2018-2025 Apostolos Chalkis
+// Copyright (c) 2025-2025 Iva Janković
+
+// Contributed and/or modified by Iva Janković, as part of Google Summer of Code 2025 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#include "doctest.h"
+#include <fstream>
+#include <iostream>
+
+#include <boost/random.hpp>
+#include <boost/random/uniform_int.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/uniform_real_distribution.hpp>
+
+#include "misc/misc.h"
+#include "random_walks/billiard_shake_and_bake_walk.hpp"
+
+#include "volume/volume_sequence_of_balls.hpp"
+#include "generators/known_polytope_generators.h"
+#include "sampling/sampling.hpp"
+
+#include "diagnostics/univariate_psrf.hpp"
+
+#include "preprocess/feasible_point.hpp"
+
+template
+<
+    typename MT,
+    typename WalkType,
+    typename Polytope
+>
+
+MT get_samples_billiard_shake_and_bake(Polytope &P)
+{
+    typedef typename Polytope::PointType Point;
+    typedef typename Polytope::NT NT;
+
+    typedef BoostRandomNumberGenerator<boost::mt19937, NT, 3> RNGType;
+
+    unsigned int nr = 3, walkL = 10, numpoints = 10000, nburns = 0, d = P.dimension();
```

Comment:
how fast is this? Can we reach psrf 1.1 with 1000 points. I am wondering whether this is to slow for a unit test.